### PR TITLE
Feat: Legacy offline manager

### PR DESCRIPTION
**Pull request by orca-nazar**
## Description

Added Legacy Offline Manager module based on this [discussion](https://github.com/rnmapbox/maps/discussions/3028).
 

## Checklist

- [x] I have tested this on a device/simulator for each compatible OS
- [x] I updated the documentation with running `yarn generate` in the root folder
- [ ] I added/updated a sample - if a new feature was implemented (`/example`)

### CODE REVIEW

**Comment by mfazekas** [comment]
@orca-nazar thanks much, I have concerns about doing it as.a build variant. 
- We have enough variants V11 and fabric, we should only add more variants only if realy-really necessary
- JS interface will of Legacy and regular is the same. So if we change the js interface we should be careful not to affect on or the other.

So I'd prefer a duplicating the JS code and having 2 component instead of conditions compilation.

What do you think?

**Comment by orca-nazar** [comment]
@mfazekas Good points. I will create offlineManagerLegacy.ts and OfflinePacksLegacy.ts as both of them will depend on the native module. Not sure about a sample in Example app and documentation. Do you think it would be good to duplicate it as well?

**Comment by orca-nazar** [comment]
@mfazekas Hey, updated PR with the new JS class for Legacy manager. It misses packs subscriptions functionality. If someone requests, I will add them as well

**Comment by mfazekas** [comment]
@orca-nazar looks good to me, can you look into the CI?

File: __tests__/modules/offline/offlineManagerLegacy.test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { NativeModules } from 'react-native';
+
+import MapboxGL from '../../../src';
```
Comment (mfazekas) [inline __tests__/modules/offline/offlineManagerLegacy.test.js:3]:
```suggestion
import Mapbox from '../../../src';
```

**Comment by orca-nazar** [comment]
> @orca-nazar looks good to me, can you look into the CI?

@mfazekas So to trigger CI, I need to get your approval. I think I fixed it

**Review by mfazekas — APPROVED** [review]

**Pull request merged**